### PR TITLE
MINOR: Remove KafkaStreams#toString

### PR DESCRIPTION
**Pull request by guozhangwang**
Remove the deprecated KafkaStreams#toString function. Also override toString() for internal classes for debugging purposes.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Do we need so many PRs for removal of deprecated code? It seems more efficient to group them a bit more...

**Comment by ijuma** [comment]
To be a bit more concrete, it could be done by done per module, per package or something like that.

**Comment by guozhangwang** [comment]
As we are still discussing which deprecated APIs to remove, it is actually better to keep it per component / KIP. And also smaller PRs are easier to review.

**Comment by ijuma** [comment]
It's actually not true that smaller PRs are always easier to review. As most things, there is a sweet spot and excessively small PRs generate more work and history noise (git blame becomes less effective for example). In any case, it was just a suggestion. If this approach works for Streams, then fine.

**Review by bbejeck — COMMENTED** [review]
Thanks, @guozhangwang, not sure if you are done with this PR, but so far LGTM.

**Comment by bbejeck** [comment]
retest this please

**Comment by guozhangwang** [comment]
@bbejeck @mjsax @vvcephei ready for reviews.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java [hunk 1]
```diff
@@ -56,6 +56,17 @@ public StreamsMetadataState(final InternalTopologyBuilder builder, final HostInf
         this.thisHost = thisHost;
     }
 
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("GlobalMetadata: " + allMetadata + "\n");
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java]: using a builder and string concat? maybe just `append("blah").append(var).append("\n");

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java [hunk 2]
```diff
@@ -445,8 +445,21 @@ void maybePurgeCommitedRecords() {
...
+     * @return A string representation of the TaskManager instance.
+     */
+    @Override
+    public String toString() {
+        return toString("");
+    }
+
     public String toString(final String indent) {
         final StringBuilder builder = new StringBuilder();
+        builder.append("TaskManager\n");
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java:461]: ditto

**Review by bbejeck — APPROVED** [review]
Minor nits aside, LGTM

**Review by mjsax — APPROVED** [review]
LGTM.

**Pull request merged**